packages/flutter/lib/src/widgets/icon.dart

```diff
@@ -231,14 +233,30 @@ class Icon extends StatelessWidget {
   /// specified, either directly using this property or using [Directionality].
   final TextDirection? textDirection;
 
+  /// Whether the size is affected by the context's text scaler.
+  ///
+  /// If true, the icon size is going to be scaled by the [TextScaler] of the
```

Comment:
nit: consider adding a sentence or two to describe common use cases / when this parameter can be helpful.

Reply:
Done